<?xml version="1.0"?>
<launch>
    <arg name="play_bag" default="true"/>
    <arg name="bag_file_name" default="$(find analysis_mp_chaser)/world/los_keeper_exp_2d.bag"/>
    <arg name="rviz_display" default="true"/>

    <group if="$(arg play_bag)">
        <node name="ros_bag_play" pkg="rosbag" type="play" args="--clock -r 1.0 -s 0.0 $(arg bag_file_name)"/>
    </group>
    <group if="$(arg rviz_display)">
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find analysis_mp_chaser)/rviz_config/los_keeper_exp_3d_total.rviz"/>
    </group>

    <node pkg="analysis_mp_chaser" type="analyzer_los_keeper" name="analyzer" output="screen">
        <param name="is_2d" value="true"/>
        <param name="is_exp" value="true"/>
        <param name="write_total_trajectory" value="false"/>
        <param name="total_trajectory_filename_write" value="$(find analysis_mp_chaser)/log/actor_trajectory_exp_3d.txt"/>
        <param name="total_trajectory_filename_read" value="$(find analysis_mp_chaser)/data/actor_trajectory_exp_3d.txt"/>
        <remap from="/los_simulator/drone_state" to="/experiment_parser/keeper_state"/>
        <remap from="/los_simulator/target_state" to="/experiment_parser/target_state"/>
        <remap from="/los_simulator/obstacle_list_state" to="/experiment_parser/obstacle_state_list"/>
        <param name="x_min" value="-4.0"/>
        <param name="x_max" value="4.0"/>
        <param name="y_min" value="-3.0"/>
        <param name="y_max" value="3.0"/>
        <param name="z_min" value="0.9"/>
        <param name="z_max" value="2.0"/>
    </node>
    <node pkg="los_keeper" type="los_keeper" name="los_keeper" output="screen">
        <rosparam command="load" file="$(find los_keeper)/param/log_plan_3d_exp_multi_thread.yaml"/>
        <rosparam command="load" file="$(find los_keeper)/param/visualization_experiment.yaml"/>
    </node>
    <node pkg="tf" type="static_transform_publisher" name="world2map" args="0 0 0 0 0 0 world map 100"/>

</launch>